resources/lib/classes.py

```diff
@@ -0,0 +1,149 @@
+import urlparse
+import urllib
+import unicodedata
+
+
+class genre(object):
+    def __init__(self):
+        self.fanart = None
+        self.thumb = None
+        self.title = None
+        self.genre_slug = None
+
+    def make_kodi_url(self):
+        d = self.__dict__
```

Comment:
shouldn't really use __dict__ like this. Are you dealing with an unknown number of options?

Reply:
Having an if statement for every class attribute eg: https://github.com/xbmc-catchuptv-au/plugin.video.abc_iview/blob/master/resources/lib/classes.py#L300 seemed very repetitive and I thought about doing it programatically, and just left the if statements if the data needed to be manipulated like url quoting. This way if any attributes get added or deleted later on the function won't necessarily have to be modified.

I've since learned that I should use vars() instead of __dict__ - would that work for you?

Reply:
Yeah, vars is perfect. I definitely agree with you, the multiple if statements is garbage and this is a nice solution. +1